### ISSUE TITLE
[clangd][HLSL] register hover not triggered when cursor is inside the argument

### ISSUE THREAD

**Issue by mafeguimaraes** [llvm/llvm-project#212749]
## Expected behavior

Hovering on the slot identifier inside `register(t1)` (e.g. on `t1`) should display the
register documentation, including a description of what `t` means (SRV).

## Observed behavior

Hovering inside the argument produces no tooltip, only hovering on the `register` keyword
itself works.

## Reproduction

```hlsl
Texture2D tex : register(t1);
```

Hovering on `register` shows the expected documentation. Hovering on `t1` (inside the
parentheses) shows nothing.

## Root cause

`ParseHLSLAnnotations` (`clang/lib/Parse/ParseHLSL.cpp`) captures a single `SourceLocation
Loc` right after consuming the `register`/annotation token, and this is the only location
passed to `Attrs.addNew`:

```cpp
Attrs.addNew(II, Loc, AttributeScopeInfo(), ArgExprs.data(), ArgExprs.size(),
             ParsedAttr::Form::HLSLAnnotation());
```

No `SourceRange` spanning the full `register(...)` construct (including its closing `)`) is
ever captured or passed, so the resulting attribute has an effectively zero-width range. When
clangd's `SelectionTree` checks whether the cursor falls inside an attribute, a zero-width
range never matches, so hover requests inside the argument never reach the attribute.

## Proposed fix

Inside the `AT_HLSLResourceBinding` case, capture the closing `)` location before it's
consumed by `ExpectAndConsume(tok::r_paren, ...)`, and thread a full `SourceRange` (from the
attribute start to the closing paren) into `Attrs.addNew` instead of the single
`SourceLocation`.
